.github/workflows/package-unit-tests.yml

```diff
@@ -19,27 +19,50 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        workspace:
-          - realm
-          - '@realm/bindgen'
-          - '@realm/network-transport'
-          - '@realm/babel-plugin'
-          - '@realm/react'
-    name: ${{ matrix.workspace }} unit tests
+        variant:
+          - {workspace: realm}
+          - {workspace: '@realm/bindgen'}
+          - {workspace: '@realm/network-transport'}
+          - {workspace: '@realm/babel-plugin'}
+          - {workspace: '@realm/react', use-baas: true}
+    name: ${{ matrix.variant.workspace }} unit tests
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
         with:
           submodules: "recursive"
+
       - uses: actions/setup-node@v3
         with:
           node-version: 18
+
       # ninja-build is used by default if available and results in faster build times
       - name: Install ninja
         run: sudo apt-get install ninja-build
+
       - name: ccache
         uses: hendrikmuhs/ccache-action@v1
+
+      - name: Generate server configuration
+        if: ${{matrix.variant.use-baas}}
+        id: baas-config
+        run:
+          suffix=$(node -p 'Math.floor(Math.random()*Number.MAX_SAFE_INTEGER)');
+          subdomain="realm-js-test-server-${{ github.run_id }}-${{ github.run_attempt }}-${suffix}";
+          echo "subdomain=${subdomain}" >> $GITHUB_OUTPUT;
+          echo "url=https://${subdomain}.ngrok.io" >> $GITHUB_OUTPUT;
+
+      - name: Trigger the test server workflow to start the server
+        if: ${{matrix.variant.use-baas}}
+        run: gh workflow run test-server.yml -f ngrok_subdomain=${{ steps.baas-config.outputs.subdomain }} -f run_id=${{ github.run_id }}
+        env:
+          GH_TOKEN: ${{ github.token }}
+
+      - name: Set baas env
+        if: ${{matrix.variant.use-baas}}
+        run: echo "realmBaseUrl=${{ steps.baas-config.outputs.url }}" >> $GITHUB_ENV
+
       # Install the root package to get dev-dependencies
       # (--ignore-scripts to avoid downloading or building the native module)
-      - run: npm ci --ignore-scripts
-      - run: npm test --workspace ${{ matrix.workspace }}
+      - run: npm ci
```

Comment:
It was needed, as it wasn't doing any of the `wireit` parts associated with running `npm install`

Reply:
Perhaps we could instead make those invocations explicit instead of relying on `postinstall` scripts on CI?
This was added to ensure we're not accidentally running tests against a prebuilt binary downloaded from NPM.

Reply:
Testing this again now and it seems the `--ignore-scripts` was not necessary.

Reply:
But have you tried making changes to the native code? I suspect that it'll download the prebuilt binary from NPM instead of building from source.

Reply:
Sorry, my previous comment was misleading.  Everything works fine with `--ignore-scripts` so I have added it back in.